packages/amplify-graphql-relational-transformer/src/schema.ts

```diff
@@ -262,6 +268,9 @@ export const ensureHasManyConnectionField = (
   const relatedTypeObject = ctx.output.getType(relatedType.name.value) as ObjectTypeDefinitionNode;
   const connectionAttributeName = getConnectionAttributeName(ctx.featureFlags, object.name.value, field.name.value, connectionFieldName);
 
+  // The nullabilty of connection fields for hasMany depends on the hasMany field
```

Comment:
I keep this as the original implementation. This is out of the scope of fixing the issue.